io_scene_godot/converters/animation.py

```diff
@@ -634,37 +685,88 @@ def export_animation_data(escn_file, export_settings, godot_node,
     """Export the action and nla_tracks in blender_object.animation_data,
     it will further call the action exporting function in AnimationDataExporter
     given by `func_name`"""
-    if (blender_object.animation_data is None or
-            not export_settings['use_export_animation']):
+    if not export_settings['use_export_animation']:
         return
-    animation_player = get_animation_player(
-        escn_file, export_settings, godot_node)
+    has_obj_cst = has_object_constraint(blender_object)
+    has_pose_cst = has_pose_constraint(blender_object)
+    need_bake = action_type == 'transform' and (has_obj_cst or has_pose_cst)
+
+    def action_baker(action_to_bake):
+        """baker for transform action, note it is a closure"""
```

Comment:
Not quite a closure. Sure, it's using the environment from the parent function but it isn't ever used outside it's parent function. If you were (for whatever reason) returning the `action_baker` function from the end of `export_animation_data`, then it would be a closure.

Reply:
fixed,  I just want to keep a note that this function use variable outside